include/rwqueue.h

```diff
@@ -44,10 +44,12 @@ class RWQueue {
 	RWQueue<T>& operator=(const RWQueue<T>& other) = delete;
 
 	RWQueue(size_t queue_capacity);
+	void Resize(size_t queue_capacity);
 
 	bool IsEmpty();
 	size_t Size();
 	size_t MaxCapacity() const;
+	int GetPercentFull();
```

Comment:
I'd use a float/double here.

Reply:
Good call.
In this case, I let the use case's needs (printing the percentage) dictate the type; but that's a flawed approach! Float it is.